CARDS-2460: Compact layout for sections

marta- opened the pull request:
**First attempt to a compact layout option for sections.**

Currently, if the section is set to have compact layout and the screen is wide enough, the entries should be displayed 3 per row (>1200px) or 2 per row (>900px). On narrower screens compact and non-compact look the same with one entry per row.

**To test:**
* Use the `Compact Sections Test` questionnaire
OR
* In Admin > Questionnaires, design a questionnaire containing a section with the `Compact` switch turned on
* Add questions to the section
* Preview the questionnaire to see the resulting layout

**Future work:**
* More flexible arrangement of entries in rows

**Known limitations:**
* The layout doesn't adjust well when there are hidden conditional entries in the compact section
* Compact sections inside compact sections probably look awful
* In patient portal, the form width is **at most 780px** - too narrow for 3/row, and even with 2/row it looks crowded for this type of user. For simplicity, the section's `compact` flag has no effect when the form is displayed in the patient portal.

marta- committed:
CARDS-2460: Compact layout for sections

marta- committed:
CARDS-2460: Compact layout for sections
Minor layout adjustments and smoother layout transition when changing the window width

sashaandjic commented:
Love it. Can we do something about the gap?
![image](https://github.com/data-team-uhn/cards/assets/23243244/c3e05f5b-315d-48b5-971e-66f2f4ffb738)

![image](https://github.com/data-team-uhn/cards/assets/23243244/95bfde30-dafa-43ab-bdf6-24cd1676c8a8)

sashaandjic commented:
Better! 

https://github.com/data-team-uhn/cards/assets/23243244/1d7dc60d-afb4-459f-a1cb-282560766dfc

marta- committed:
CARDS-2460: Compact layout for sections
Reduced the input width in compact sections on narrow screens

sashaandjic reviewed (approved):
tested

marta- committed:
CARDS-2460: Compact layout for sections
Quick fix for patient portal layout issues - disable compact sections in patient portal, since the available with is reduced.

marta- committed:
CARDS-2460: Compact layout for sections
Added test questionnaire created by @sashaandjic

sashaandjic reviewed (approved):
tested - looks awesome

There is one really minor thing that should not prevent this PR from merging.
In view mode: due to the missing info box in section `Your favorites` there is a gap between `fav food` and `fav drink`:
![image](https://github.com/data-team-uhn/cards/assets/23243244/8fc34848-731d-4e17-b375-ab6b49efe8ae)

acrowthe reviewed (approved):
Tested briefly, code looks good

marta- commented:
Rebased to resolve conflicts after the merge of #1676 .

sdumitriu commented on modules/data-model/forms/api/src/main/resources/SLING-INF/nodetypes/forms.cnd:
```diff
@@ -310,6 +310,9 @@
   // Section display mode, one of default, header, footer, matrix
   - displayMode (string) = 'default' autocreated
 
+  // For with the default display mode, whether to use horizontal layout (compact = true) to minimize negative space, or use vertival layour (default)
```
For with?

marta- replied:
```suggestion
  // For sections with the default display mode, whether to use horizontal layout (compact = true) to minimize negative space, or use vertical layout (default)
```

marta- committed:
CARDS-2460: Compact layout for sections

Typo fixes

sdumitriu commented:
There is still a gap in the middle in view mode when not all of the items present in edit mode show up, like:
![image](https://github.com/data-team-uhn/cards/assets/88663/3cedc962-3c25-4f00-8cef-c11e6804d33a)
